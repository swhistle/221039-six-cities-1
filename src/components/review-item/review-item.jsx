import React from "react";
import PropTypes from "prop-types";

export class ReviewItemComponent extends React.PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    const {review} = this.props;

    return <React.Fragment>
      <div className="reviews__user user">
        <div className="reviews__avatar-wrapper user__avatar-wrapper">
          <img className="reviews__avatar user__avatar" src={review.user.avatar_url} width="54" height="54" alt="Reviews avatar"/>
        </div>
        <span className="reviews__user-name">
          {review.user.name}
        </span>
      </div>
      <div className="reviews__info">
        <div className="reviews__rating rating">
          <div className="reviews__stars rating__stars">
            <span style={{width: `94%`}}></span>
            <span className="visually-hidden">Rating</span>
          </div>
        </div>
        <p className="reviews__text">
          {review.user.comment}
        </p>
        <time className="reviews__time" dateTime="2019-04-24">{review.user.date}</time>
      </div>
    </React.Fragment>;
  }
}

ReviewItemComponent.propTypes = {
  review: PropTypes.shape({
    id: PropTypes.number,
    user: PropTypes.object,
    rating: PropTypes.number,
    comment: PropTypes.string,
    dateTime: PropTypes.string
  })
};
