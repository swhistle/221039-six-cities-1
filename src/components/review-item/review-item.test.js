import React from 'react';
import renderer from 'react-test-renderer';

import {ReviewItemComponent} from "./review-item.jsx";

const TEST_REVIEW_ITEM = {
  id: 1,
  rating: 4,
  user: {
    id: 2,
    name: `user`
  },
  comment: `very good`,
  dateTime: `2019-05-08T14:13:56.569Z`
};

it(`ReviewItemComponent correct renders`, () => {
  const reviewItem = renderer
    .create(
        <ReviewItemComponent review={TEST_REVIEW_ITEM}/>
    ).toJSON();

  expect(reviewItem).toMatchSnapshot();
});
