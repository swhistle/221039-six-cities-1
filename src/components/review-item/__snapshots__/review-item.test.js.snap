// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewItemComponent correct renders 1`] = `
Array [
  <div
    className="reviews__user user"
  >
    <div
      className="reviews__avatar-wrapper user__avatar-wrapper"
    >
      <img
        alt="Reviews avatar"
        className="reviews__avatar user__avatar"
        height="54"
        width="54"
      />
    </div>
    <span
      className="reviews__user-name"
    >
      user
    </span>
  </div>,
  <div
    className="reviews__info"
  >
    <div
      className="reviews__rating rating"
    >
      <div
        className="reviews__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "80%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <p
      className="reviews__text"
    >
      very good
    </p>
    <time
      className="reviews__time"
    >
      Invalid Date
    </time>
  </div>,
]
`;
